build.gradle

```diff
@@ -188,19 +185,27 @@ task download(type: Copy) {
   into 'src/jar'
 }
 
-task gitRevList(type: Exec) {
+task gitRevList() {
   onlyIf {
     file('.git').exists()
   }
   inputs.dir('.git')
   outputs.files file('build/revision.txt')
-  commandLine 'git', 'rev-list', '--count', 'origin/main'
-  standardOutput = new ByteArrayOutputStream()
 
   doLast {
-    file('build/revision.txt').text = standardOutput.toString().trim()
+    def revision = grgit.log(includes:['HEAD']).size()
+    file('build/revision.txt').text = revision.toString().trim()
     // Update the version to the new revision
-    project.setVersion(standardOutput.toString().trim())
+    project.setVersion(revision.toString().trim())
+  }
+}
+
+task gitUpdate() {
+  doLast {
+  grgit.pull {
+    remote= 'origin'
+    branch= 'main'
```

Comment:
Would prefer if this updated the current branch instead.

Maybe check for dirty state, and stash/pull/stash pop

Would a rebase command be useful or overwhelming?

Reply:
So, the pull command also has a boolean for rebase which we could use as well.

Reply:
Sadly, grgit does not have stash support.

Also, spotless isn't formatting this config well, clearly. Hm.

Reply:
Is there an option to print some sort of status?  Right now it acts silently, which isn't super great for knowing if everything is up to date.

Reply:
seeing that now, thank you.